src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1025,37 +1024,88 @@ protected static GenotypesContext updateGenotypesWithMappedAlleles(final Genotyp
         return updatedGenotypes;
     }
 
-    public static int computeReverseClipping(final List<Allele> unclippedAlleles, final byte[] ref) {
-        int clipping = 0;
-        boolean stillClipping = true;
+    /**
+     * Clip out any unnecessary bases off the front of the alleles
+     *
+     * The VCF spec represents alleles as block substitutions, replacing AC with A for a
+     * 1 bp deletion of the C.  However, it's possible that we'd end up with alleles that
+     * contain extra bases on the left, such as GAC/GA to represent the same 1 bp deletion.
+     * This routine finds an offset among all alleles that can be safely trimmed
+     * off the left of each allele and still represent the same block substitution.
+     *
+     * A/C => A/C
+     * AC/A => AC/A
+     * ACC/AC => CC/C
+     * AGT/CAT => AGT/CAT
+     * <DEL>/C => <DEL>/C
+     *
+     * @param alleles a non-null list of alleles that we want to clip
+     * @param isReverse if true, clip from the end of the allele if false, clip from the beginning of the allele.
+     * When isReverse is true:
+     * AC/A => AC/A
+     * AT/AT => A/A
+     * CCG/CG => CC/C
+     * TGCC/TGCC => T/T
+     * When isReverse is false:
+     * AC/A => AC/A
+     * ACG/A => ACG/A
+     * AGG/AG => GG/G
+     * ACG/AC => CG/C
+     * G/GGGGG => G/GGGGG
+     * @return the offset into the alleles where we can safely clip, inclusive, or
+     *   -1 if no clipping is tolerated.  So, if the result is 0, then we can remove
```

Comment:
This "-1 if no clipping is tolerated" is not needed any more.  Also, if the result is 0, we can't remove any bases, and if the result is 1, we can remove one base.